app/controllers/holds_controller.rb

```diff
@@ -122,4 +178,36 @@ def past_date
 
       redirect_to holds_path
     end
+
+    def hold_lookup(hold_key)
+      hold_record = symphony_client.get_hold_info hold_key, current_user.session_token
+      parsed_hold = JSON.parse hold_record.body
+      Hold.new parsed_hold
+    end
+
+    def item_lookup(barcode)
+      item_record = symphony_client.get_item_info barcode, current_user.session_token
+      parsed_item = JSON.parse item_record.body
+      Item.new parsed_item
+    end
+
+    def authorize_create!
+      return unless missing_args?
+
+      raise HoldCreateException, 'Error'
+    end
+
+    def deny_create
+      flash[:error] = if missing_args?
+                        t 'myaccount.hold.place_hold.missing_params'
+                      else
+                        t 'myaccount.hold.place_hold.select_volumes'
+                      end
+
+      redirect_to new_hold_path(catkey: params[:catkey])
+    end
+
+    def missing_args?
+      params['pickup_library'] === 'Not set' || params['pickup_by_date'].blank?
```

Comment:
```suggestion
      params['pickup_library'] == 'Not set' || params['pickup_by_date'].blank?
```

So, using `===` kinda threw me here. The [String::=== method](https://ruby-doc.org/core-2.6/String.html#method-i-3D-3D-3D) just returns `==` with the added side affect of also passing true if the left side of this comparison is `String`.

Here's a weird thing that wouldn't happen but could:

```
irb(main):009:0> 'y'.class === 'x'
=> true
```

In JavaScript `===` is often preferred because `==` coerces to string, but ruby doesn't coerce at the `==` level of comparing. Also, looks like [eql? works the same as == in ruby](https://ruby-doc.org/core-2.6.3/Object.html#method-i-eql-3F).

```
irb(main):016:0> '1' == 1
=> false
irb(main):018:0> '1'.eql? 1
=> false
```

Sorry, too much on one little thing, just kinda threw me a bit.

Reply:
I do have a memory of why and how I used `===` here :) Yup, your point makes sense 👍

Reply:
And sorry again, not trying to be nitpicky or pedantic, I just kinda got down the rabbit hole on this one 🤣